fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -92,13 +93,18 @@
 @API(API.Status.EXPERIMENTAL)
 public class LuceneIndexMaintainer extends StandardIndexMaintainer {
     private static final Logger LOG = LoggerFactory.getLogger(LuceneIndexMaintainer.class);
+    private static final long SERIALIZER_LOG_DELAY = TimeUnit.SECONDS.toMillis(3);
+
     private final FDBDirectoryManager directoryManager;
     private final LuceneAnalyzerCombinationProvider indexAnalyzerSelector;
     private final LuceneAnalyzerCombinationProvider autoCompleteAnalyzerSelector;
     protected static final String PRIMARY_KEY_FIELD_NAME = "_p";
     protected static final String PRIMARY_KEY_SEARCH_NAME = "_s";
     protected static final String PRIMARY_KEY_BINARY_POINT_NAME = "_b";
     private final Executor executor;
+    LuceneIndexKeySerializer keySerializer;
+    // The last time (in System.currentTimeMillis) this instance logged a serializer error
```

Comment:
I was concerned with thousands of messages coming down in every transaction (say when lots of documents get written in a single transaction), in case there is such a use case.
Would you think the at once-per-transaction is strong enough of a protection?

Reply:
My point was more that there is one index maintainer per transaction, so you have a limit of once every 3 seconds, but a transaction is only 5 seconds, so at most this protection (as you have it) is that it will log at most twice per transaction, so you could simplify the code to just have a boolean with basically the same protection.

Reply:
Yeah, that makes sense, and because of the way things are set up we can either make this guard a static variable (which I don't like much), add another random scaling on top of the log call (say, one in a hundred), or just disable logging for this class at runtime if there are too many logs (which would have to be resolved quickly anyways since the index maintainer is not working right).
I'll modify to have this be a boolean, and rely on the log management to stop logging in case there are too many messages.

Reply:
Done.